sdk/webpubsub/azure-webpubsub-client/azure/webpubsub/client/_client.py

```diff
@@ -0,0 +1,835 @@
+    :keyword float reconnect_retry_backoff_max: The maximum back off time. Default value is 120.0 seconds
+    :keyword RetryMode reconnect_retry_mode: Fixed or exponential delay between attemps, default is exponential.
+    :keyword int message_retry_total: total number of retries to allow for sending message. Default is 3.
+    :keyword float message_retry_backoff_factor: A backoff factor to apply between attempts after the second try
+     (most errors are resolved immediately by a second try without a delay). In fixed mode, retry policy will always
+     sleep for {backoff factor}. In 'exponential' mode, retry policy will sleep for:
+     "{backoff factor} * (2 ** ({number of retries} - 1))" seconds. If the backoff_factor is 0.1, then the
+     retry will sleep for [0.0s, 0.2s, 0.4s, ...] between retries. The default value is 1.0.
+    :keyword float message_retry_backoff_max: The maximum back off time. Default value is 120.0 seconds
+    :keyword RetryMode message_retry_mode: Fixed or exponential delay between attemps, default is exponential.
+    :keyword bool auto_rejoin_groups: auto_rejoin_groups, default is True
+    :keyword str user_agent: The user agent to be used for the request. If specified, this will be added in front of
+     the default user agent string.
+    :keyword bool logging_enable: Whether to output network trace logs to the logger. Default is `False`.
+    :keyword float ack_timeout: Time limit to wait for ack message from server. The default value is 30.0 seconds.
+    :keyword float start_timeout: Time limit to wait for successful client start. The default value is 30.0 seconds.
+    """
+
+    def __init__(
+        self,
+        credential: Union[WebPubSubClientCredential, str],
+        **kwargs: Any,
+    ) -> None:
+        if isinstance(credential, WebPubSubClientCredential):
+            self._credential = credential
+        elif isinstance(credential, str):
+            self._credential = WebPubSubClientCredential(credential)
+        else:
+            raise TypeError("type of credential must be str or WebPubSubClientCredential")
+        reconnect_retry_total = kwargs.pop("reconnect_retry_total", _RETRY_TOTAL)
+        self._auto_reconnect = reconnect_retry_total > 0
+        self._auto_rejoin_groups = kwargs.pop("auto_rejoin_groups", True)
+        protocol_type = kwargs.pop("protocol_type", WebPubSubProtocolType.JSON_RELIABLE)
+        protocol_map = {
+            WebPubSubProtocolType.JSON: WebPubSubJsonProtocol,
+            WebPubSubProtocolType.JSON_RELIABLE: WebPubSubJsonReliableProtocol,
+        }
+        if protocol_type in protocol_map:
+            self._protocol = protocol_map[protocol_type]()
+        else:
+            self._protocol = WebPubSubJsonReliableProtocol()
+
+        self._reconnect_retry_policy = RetryPolicy(
+            retry_total=reconnect_retry_total,
+            retry_backoff_factor=kwargs.pop("reconnect_retry_backoff_factor", _RETRY_BACKOFF_FACTOR),
```

Comment:
@annatisch just to make sure we're ok with the names for these public kwargs

Reply:
I have discussed with anna about the name and we made an agreement with pattern `"reconnect/recovery_" + retry_variables of azure-core`

Reply:
erm I don't recall that naming pattern... I don't think we should have _both_ "reconnect/recovery" and "retry" - these mean the same thing. I think we should use _only_ one of these verbs where it makes sense to do so.

Reply:
hmm actually I've refreshed myself on the context - disregard my previous comment!
These names are fine for now - Will have a think if there's a better way to represent this.